### PR TITLE
AttackIQ enhancement

### PR DESCRIPTION
**Pull request by orlichter1**
## Status
Ready 
## Related Issues
fixes: https://github.com/demisto/etc/issues/19345

## Description
Enhancement. includes the following commands:
* attackiq-list-assessment-templates
* attackiq-list-assets
* attackiq-create-assessment
* attackiq-add-assets-to-assessment

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [x] [Documentation](https://github.com/demisto/etc/issues/19773)
- [ ] Code Review

## Technical writer review
Mention and link to the files that require a technical writer review.
- [ ] [YAML file](https://github.com/demisto/content/edit/AttackIQ/Integrations/AttackIQFireDrill/AttackIQFireDrill.yml?pr=%2Fdemisto%2Fcontent%2Fpull%2F4546)
- [ ] [CHANGELOG](https://github.com/demisto/content/edit/AttackIQ/Integrations/AttackIQFireDrill/CHANGELOG.md?pr=%2Fdemisto%2Fcontent%2Fpull%2F4546)

### CODE REVIEW

File: Integrations/AttackIQFireDrill/AttackIQFireDrill.py [hunk 1]
```diff
@@ -122,9 +122,11 @@
 ''' HELPER FUNCTIONS '''
 
 
+@logger
 def http_request(method, url_suffix, params=None, data=None):
     url = f'{SERVER}/{url_suffix}'
-    LOG(f'attackiq is attempting {method} request sent to {url} with params:\n{json.dumps(params, indent=4)}')
+    LOG(f'AttackIQ is attempting {method} request sent to {url} with params:\n{json.dumps(params, indent=4)} \n '
```
Comment (yaakovi) [inline Integrations/AttackIQFireDrill/AttackIQFireDrill.py]: no need for both `@logger` and this line.

File: Integrations/AttackIQFireDrill/AttackIQFireDrill.py [hunk 2]
```diff
@@ -135,6 +137,8 @@ def http_request(method, url_suffix, params=None, data=None):
             headers=HEADERS
         )
         # Handle error responses gracefully
+        if res.status_code == 204:
```
Comment (yaakovi) [inline Integrations/AttackIQFireDrill/AttackIQFireDrill.py]: move before the comment as this condition is not an error

File: Integrations/AttackIQFireDrill/AttackIQFireDrill.yml [hunk 3]
```diff
@@ -481,16 +484,212 @@ script:
...
+    description: List all assets.
+    execution: false
+    name: attackiq-list-assets
+    outputs:
+    - contextPath: AttackIQ.Asset.ID
+      description: The asset ID.
+      type: Unknown
+    - contextPath: AttackIQ.Asset.Description
+      description: The description of the asset.
+      type: Unknown
```
Comment (yaakovi) [inline Integrations/AttackIQFireDrill/AttackIQFireDrill.yml]:
```suggestion
      type: String
```

File: Integrations/AttackIQFireDrill/AttackIQFireDrill.yml [hunk 4]
```diff
@@ -481,16 +484,212 @@ script:
...
+      description: Is the template hidden.
+      type: Boolean
+  - deprecated: false
+    description: List all assets.
+    execution: false
+    name: attackiq-list-assets
+    outputs:
+    - contextPath: AttackIQ.Asset.ID
+      description: The asset ID.
+      type: Unknown
```
Comment (yaakovi) [inline Integrations/AttackIQFireDrill/AttackIQFireDrill.yml]:
```suggestion
      type: String
```

File: Integrations/AttackIQFireDrill/AttackIQFireDrill.py [hunk 5]
```diff
@@ -525,6 +529,137 @@ def run_all_tests_in_assessment_command():
...
+        raise ValueError(f"Could not create an assessment. Specifically: {str(e)}")
+
+    assessment_id = res.get('project_id')
+    try:
+        raw_assessments = get_assessments(assessment_id=assessment_id)
+        assessments_res = build_transformed_dict(raw_assessments, ASSESSMENTS_TRANS)
+        hr = tableToMarkdown(f'Created Assessment: {assessment_id} successfully.', assessments_res,
+                             headers=['Id', 'Name', 'Description', 'User', 'Created', 'Modified'])
+        return_outputs(hr, {'AttackIQ.Assessment(val.Id === obj.Id)': assessments_res}, raw_assessments)
+    except HTTPError as e:
```
Comment (yaakovi) [inline Integrations/AttackIQFireDrill/AttackIQFireDrill.py]: no need to catch it here, you have it in the execution code try-except.

File: Integrations/AttackIQFireDrill/AttackIQFireDrill.py [hunk 6]
```diff
@@ -525,6 +529,137 @@ def run_all_tests_in_assessment_command():
...
+    if asset_groups:
+        data['asset_groups'] = asset_groups
+
+    if data == {}:
+        raise ValueError(f"No asset or asset groups were specified.")
+    try:
+        res = http_request('POST', f'/v1/assessments/{assessment_id}/update_defaults', data=json.dumps(data))
+        demisto.results(res.get('message', ''))
+    except Exception:
+        raise ValueError("Could not find either the assessment or one of the assets/asset groups.")
```
Comment (yaakovi) [inline Integrations/AttackIQFireDrill/AttackIQFireDrill.py]: could there be some other reason for the error?

File: Integrations/AttackIQFireDrill/AttackIQFireDrill.py [hunk 7]
```diff
@@ -525,6 +529,137 @@ def run_all_tests_in_assessment_command():
...
+def delete_assessment_command():
+    assessment_id = demisto.args().get('assessment_id')
+    try:
+        http_request('DELETE', f'/v1/assessments/{assessment_id}')
+        demisto.results(f"Deleted assessment {assessment_id} successfully.")
+    except Exception as e:
+        if '403' in str(e):
+            raise ValueError(f"Could not find the assessment {assessment_id}")
+        else:
+            raise e
```
Comment (yaakovi) [inline Integrations/AttackIQFireDrill/AttackIQFireDrill.py]:
```suggestion
            raise
```

File: Integrations/AttackIQFireDrill/AttackIQFireDrill.py [hunk 8]
```diff
@@ -525,6 +529,137 @@ def run_all_tests_in_assessment_command():
...
+
+
+@logger
+def delete_assessment_command():
+    assessment_id = demisto.args().get('assessment_id')
+    try:
+        http_request('DELETE', f'/v1/assessments/{assessment_id}')
+        demisto.results(f"Deleted assessment {assessment_id} successfully.")
+    except Exception as e:
+        if '403' in str(e):
```
Comment (yaakovi) [inline Integrations/AttackIQFireDrill/AttackIQFireDrill.py]: 403 is forbidden, did you mean 404?

File: Integrations/AttackIQFireDrill/AttackIQFireDrill.py (hunk 7 above)
Comment (orlichter1) [reply Integrations/AttackIQFireDrill/AttackIQFireDrill.py]: "raise" raises the exception? (e is redundant here, not forgotten?)

File: Integrations/AttackIQFireDrill/AttackIQFireDrill.py (hunk 8 above)
Comment (orlichter1) [reply Integrations/AttackIQFireDrill/AttackIQFireDrill.py]: no, it's 403. 🤷‍♂

**Comment by yaakovi** [comment]
@kirbles19 
please review the yml (added several new commands) and the changelog

**Comment by kirbles19** [comment]
@orlichter1 Done.

**Pull request merged**